### PR TITLE
Display delivery method in orders and orders details pages

### PR DESCRIPTION
**Pull request by j0sv**
## PT: https://www.pivotaltracker.com/story/show/151122486 ##

- Display delivery method in orders and orders details pages

### CODE REVIEW

File: app/controllers/orders_controller.rb [hunk 1]
```diff
@@ -25,8 +25,8 @@ def destroy
 
   def update
     @order = get_order
-
     if @order.update(order_params)
+      @order.update(delivery_method: params[:delivery_method])
```
Comment (tochman) [inline app/controllers/orders_controller.rb]: Is there no change in the feature tests that cover this new code?

Comment (j0sv) [reply app/controllers/orders_controller.rb]: @tochman There seems to be a problem with the method order_params, delivery_method is added to the permitted params but the delivery_method is still not saved in by "@order.update(order_params)". This is a hack to fix that. Can you see a better way to do this? 
I have added a test.

Comment (tochman) [reply app/controllers/orders_controller.rb]: What problem is that?

Comment (tochman) [reply app/controllers/orders_controller.rb]: Is the delivery method part of order_params? Perhaps it needs to be moved into the form block?

Comment (j0sv) [reply app/controllers/orders_controller.rb]: It is but it is saying that it is not permitted then I use pry to debug.

Comment (tochman) [reply app/controllers/orders_controller.rb]: Okay, I'll pull it down and have a look

Comment (j0sv) [reply app/controllers/orders_controller.rb]: ops.. it not within the form. I will fix. 
Edit: This was wrong, it is in the form.

Comment (tochman) [reply app/controllers/orders_controller.rb]: I don't see that in the `order_params` 
```ruby
order_params
=> <ActionController::Parameters {"delivery_date"=>"2017-11-10 12:00", "delivery_name"=>"Hungry corp Inc", "delivery_address"=>"Street 42", "delivery_postal_code"=>"123 45", "delivery_city"=>"Town", "delivery_floor"=>"3", "delivery_door_code"=>"1111", "delivery_contact_name"=>"John Doe", "delivery_contact_phone_number"=>"555 123 45 67", "billing_name"=>"John Doe", "billing_company"=>"Hungry corp Inc", "billing_org_nr"=>"19210713-1444", "billing_address"=>"Street 42", "billing_postal_code"=>"123 45", "billing_city"=>"Town", "billing_phone"=>"555 123 55 11", "billing_email"=>"invoice@hungrycorp.com", "allergies"=>"", "status"=>"submitted"} permitted: true>
```

```
[3] pry(#<OrdersController>)> order_params[:delivery_method]
=> nil
```

Comment (tochman) [reply app/controllers/orders_controller.rb]: You are using `form_tag` syntax in a `form_for` (!!) That's why these fields are not included in the params for order.

```
  <div class="float-left order-container">
    <h4><%= t('order.delivery_info_header') %>:</h4>
    <%= f.datetime_field :delivery_date, placeholder: t('delivery_date'), class: 'input-wide' %>
    <br>
    <%= radio_button_tag(:delivery_method, 'pickup') %>
    <%= label_tag(:delivery_method_pickup, t('order.delivery_pickup').html_safe ) %>
    <br>
    <%= radio_button_tag(:delivery_method, 'delivery', checked: true) %>
    <%= label_tag(:delivery_method_delivery, t('order.delivery_deliver').html_safe ) %>
  </div>
```

Comment (j0sv) [reply app/controllers/orders_controller.rb]: It is fixed now.

**Pull request merged**